src/main/java/harmony/dev/harmonyserver/service/MemberService.java

```diff
@@ -1,48 +1,67 @@
 package harmony.dev.harmonyserver.service;
 
+import harmony.dev.harmonyserver.Exception.ExceptionSummary;
+import harmony.dev.harmonyserver.Exception.BusinessException;
 import harmony.dev.harmonyserver.domain.Member;
 import harmony.dev.harmonyserver.repository.MemberRepository;
-import lombok.RequiredArgsConstructor;
-import org.springframework.data.util.Pair;
+
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 import javax.transaction.Transactional;
+
 import java.util.List;
+import java.util.Map;
 
 @Service
-@Transactional
-@RequiredArgsConstructor
+@Transactional  // FIXME: Move to each transaction method
 public class MemberService {
     private final MemberRepository memberRepository;
 
-    public Pair<Boolean, String> join(Member member) {
-        Pair<Boolean, String> resultPair = validateDuplicateMember(member);
-
-        if (resultPair.getFirst()){
-            memberRepository.save(member);
-        }
-        return resultPair;
+    @Autowired
+    public MemberService(MemberRepository memberRepository) {
+        this.memberRepository = memberRepository;
     }
 
-    private Pair<Boolean, String> validateDuplicateMember(Member member) {
-        if (!memberRepository.findByUserId(member.getUserId()).isEmpty()){
-            return Pair.of(false, "userId");
-        }else if(!memberRepository.findByPhoneNumber(member.getPhoneNumber()).isEmpty()){
-            return Pair.of(false, "phoneNumber");
-        }else{
-            return Pair.of(true, "ok");
-        }
-    }
+    public List<Member> getMembers(Map<String, String> params) {
+        String userId = params.get("userId");
+        String phoneNumber = params.get("phoneNumber");
 
-    public List<Member> findMembers(){
-        return memberRepository.findAll();
+        if (userId == null && phoneNumber == null) {
+            BusinessException e = new BusinessException();
+            e.add(ExceptionSummary.builder()
+                            .field("userId or phoneNumber")
+                            .code("NotFound")
+                            .message("Parameter Not Found")
```

Comment:
특정 field가 아닌 전반적인 문제라는 점에서 field를 null으로 두는 것은 어떤지 검토 바랍니다.
현재 구현도 크게 문제는 없을 것으로 보입니다.
```suggestion
                            .code("NotFound")
                            .message("No parameter given")
```

Reply:
반영해도 좋을 것 같습니다.
이름도 field 이기 때문에 message에 힌트를 주는 것으로 충분해보입니다.